### DISCUSSION TITLE
showing the last run status in the scheduled table

### DISCUSSION THREAD

**Discussion by dheise-rheinenergie-trading-com** [jhuckaby/Cronicle#492] [Show and tell]
Today I had to manually rerun a lot of failed events. Here it would have been helpful to see the run status of the last run of each event in the scheduled table. So I had an idea for a little trick to show this status.

For this I use my already running webhook listener (which collects the completed messages for my monitoring system from all of my events) to change the title of the event in dependency of the return code (using the api call update_event).

The result is something like this:

![image](https://user-images.githubusercontent.com/73586581/166252885-14b7be03-b378-42ee-9fcb-1b79e68e5d28.png)

with these nice side effects:

- I can see the run status of all my jobs at a glance
- Jobs are sorted according to their run status (error on top, successful jobs below)
- I can use the search to find all failed jobs

![image](https://user-images.githubusercontent.com/73586581/166253517-6991d761-065a-4138-acf8-1d0a5a20af69.png)

we find this pretty handy.

maybe someone can also use this idea...

Daniel

**Comment by jhuckaby** [comment]
This is a really clever idea!  I love it!  Very innovative.  I'll add this as a core feature in the next release (giving you full credit of course).

**Reply by dheise-rheinenergie-trading-com** [reply]
Hi Joseph,
thank you for your appreciation! It would be great if the feature was adopted and made available to all users.

Daniel

**Comment by mikeTWC1984** [comment]
Yeah, good idea. Jenkins has something like this, always thought it would be useful for cronicle. Actually things like that are quite simple to implement, there is already a mechanism to update event via job (by populating job.update_event property). I guess it would be beneficial to add some more info about last execution, e.g. exit code, jobid and duration. Maybe even keep last N executions. 

I've been working on a similar feature for a job to emit short  message, so it could be displayed on active job table (similar to progress), and then the last message will be kept on event, so following run would know where previous job stopped. I'll share my findings later.

One more feature from this category just came to my mind - if we can calculate average duration of previous jobs then we can estimate current job progress and auto-generate progress bar

**Reply by dheise-rheinenergie-trading-com** [reply]
Sounds really exciting. T he only question is whether there is enough room for it in the scheduled table and whether it won't be too crowded.

**Reply by jhuckaby** [reply]
Mike, you are an ocean of great ideas.  I wish I had the time to implement them all!

**Reply by dheise-rheinenergie-trading-com** [reply]
@mikeTWC1984, I installed your latest release 1.5.9 (mainly for the filtering fix in the scheduled page from @jhuckaby) and was totally surprised, that you already implemented the the status light for every job inclusive a filtering option and a collapsing option for categories! But here it is unfortunately so that the categories open again when the page refreshes. But the status lights are really great!

but for now: big thanks to both of you.

**Comment by jhuckaby** [comment]
Check out [version 0.9.7](https://github.com/jhuckaby/Cronicle/releases/tag/v0.9.7) guys.  It has this feature built in!

<img width="379" alt="Screen Shot 2022-05-04 at 9 35 09 PM" src="https://user-images.githubusercontent.com/700765/166864523-59a6c3bd-4db4-442d-82fb-7d3f56e426a1.png">

You can also filter the event list based on the last run status (in the "All Events" menu), and also click on the last run status bubbles to jump to the latest job detail page.

I should say here that this branch of Cronicle (v0) is in maintenance mode, and I usually never add new features.  The reason is, I'm hard at work over on Cronicle v2 (Orchestra), where I will have to code this feature all over again.  But this one was really low hanging fruit, easy to implement, and is such a good idea, I really couldn't resist.  Thanks @dheise-rheinenergie-trading-com!

**Reply by Bens-ct** [reply]
Like the new feature, however I do miss having the status column. It is now hard to see what jobs are actually running when you have several hundred of them.

**Reply by JorgeMega** [reply]
> Echa un vistazo a [la versión 0.9.7](https://github.com/jhuckaby/Cronicle/releases/tag/v0.9.7), chicos. ¡Tiene esta función integrada!
> 
> <img alt="Captura de pantalla 2022-05-04 a las 9 35 09 PM" width="379" src="https://user-images.githubusercontent.com/700765/166864523-59a6c3bd-4db4-442d-82fb-7d3f56e426a1.png">
> 
> También puede filtrar la lista de eventos en función del estado de la última ejecución (en el menú "Todos los eventos"), y también hacer clic en las burbujas de estado de la última ejecución para saltar a la última página de detalles del trabajo.
> 
> Debo decir aquí que esta rama de Cronicle (v0) está en modo de mantenimiento, y por lo general nunca agrego nuevas características. La razón es que estoy trabajando duro en Cronicle v2 (Orquesta), donde tendré que codificar esta función de nuevo. Pero este era una fruta muy baja, fácil de implementar, y es una buena idea, realmente no pude resistirme. ¡Gracias @dheise-rheinenergie-trading-com!

I also agree, in that, before I could see how many processes were running, now you have to go to home to see them, if it could be implemented again it would be greatly appreciated, thanks
